### DISCUSSION TITLE
Invalid API key

### DISCUSSION THREAD

**Discussion by TByte2013** [datalust/seq-tickets#1842] [Help] [answered]
I'm attempting to identify the application that's logging in with an invalid API key, but the information provided by Seq is inadequate. Seq only logs the IP address, and the first four characters of the invalid key are not sufficient to locate the application. To determine the component and environment from which the log originated, I need to access the log request body. Is there a way to view the request body in Seq?

**Comment by nblumhardt** [comment]
Hi Tom, thanks for dropping us a line. I'm AFK but I have a feeling we made some changes in this area in 2023.1 - are you running that version? Thanks!

**Reply by TByte2013** [reply]
Hi @nblumhardt, we're using 2023.1.9229 at Michael Hill, below is a snapshot of the Ingestion Log

![image](https://github.com/datalust/seq-tickets/assets/6309106/27622102-88dc-4b13-8f98-bcc7aa4cd123)

**Reply by nblumhardt** [reply, marked as answer]
Thanks for the follow up, @TByte2013.

I had a look into this - the easiest way to collect full payloads for the unidentified events is to:

1. Go to _Data > Ingestion_, click on the _Unauthenticated_ HTTP input and add an attached property such as `Origin = Unauthenticated`.
2. Go to _Settings > API keys_ and deselect _Require an API key for HTTP/S ingestion_
3. ...wait for some events to arrive...
4. Re-enable _Require an API key for HTTP/S ingestion_
5. Search in _Events_ for `Origin = 'Unauthenticated'`

Let me know if this does the trick for you.

Best regards,
Nick

**Reply by TByte2013** [reply]
Awesome, I'll get it a shot tonight. Cheers

**Reply by TByte2013** [reply]
Thank you @nblumhardt, following your steps, we're able to capture the unauthenticated event and track down the app using the invalid api key.
